Add a NeoN banner and build summary

greole opened the pull request:
This PR adds a basic banner to the cmake output. Additionally, it writes a file with a summary (see below).

Additional changes, renamed options that define c macros to ` NeoN_DEFINE `

```
+-------------------------------------------------------------------------------+                                                                                            
                                                                                                                                                                             
                                                                                                                                                                             
         b.             8 8 8888888888       ,o888888o.     b.             8                                                                                                 
         888o.          8 8 8888          . 8888     `88.   888o.          8                                                                                                 
         Y88888o.       8 8 8888         ,8 8888       `8b  Y88888o.       8                                                                                                 
         .`Y888888o.    8 8 8888         88 8888        `8b .`Y888888o.    8                                                                                                 
         8o. `Y888888o. 8 8 888888888888 88 8888         88 8o. `Y888888o. 8                                                                                                 
         8`Y8o. `Y88888o8 8 8888         88 8888         88 8`Y8o. `Y88888o8                                                                                                 
         8   `Y8o. `Y8888 8 8888         88 8888        ,8P 8   `Y8o. `Y8888                                                                                                 
         8      `Y8o. `Y8 8 8888         `8 8888       ,8P  8      `Y8o. `Y8                                                                                                 
         8         `Y8o.` 8 8888          ` 8888     ,88'   8         `Y8o.`                                                                                                 
         8            `Yo 8 888888888888     `8888888P'     8            `Yo                                                                                                 
                                                                                                                                                                             
         Version: 0.1.0 Git commit: v0.1.0-647-g4463098a4                                                                                                                    
                                                                                                                                                                             
+-------------------------------------------------------------------------------+                                                                                            
 Build information:                                                                                                                                                          
--      CMAKE_BUILD_TYPE = RelWithDebInfo                                                                                                                                    
--      CMAKE_CXX_COMPILER = GNU@14.2.1                                                                                                                                      
+-------------------------------------------------------------------------------+                                                                                            
 Third-Party dependencies:                                                                                                                                                   
 --     NeoN_WITH_ADIOS2 = OFF                                                                                                                                               
 --     NeoN_WITH_GINKGO = ON                                                                                                                                                
 --     NeoN_WITH_PETSC = OFF                                                                                                                                                
 --     NeoN_WITH_SUNDIALS = ON                                                                                                                                              
+-------------------------------------------------------------------------------+                                                                                            
 NeoN components:                                                                                                                                                            
 --     NeoN_BUILD_BENCHMARKS = ON                                                                                                                                           
 --     NeoN_BUILD_DOC = OFF                                                                                                                                                 
 --     NeoN_BUILD_TESTS = OFF                                                                                                                                               
+-------------------------------------------------------------------------------+                                                                                            
 NeoN definitions:                                                                                                                                                           
 --     NeoN_DEFINE_DP_LABEL = OFF                                                                                                                                           
 --     NeoN_DEFINE_DP_SCALAR = ON                                                                                                                                           
 --     NeoN_DEFINE_US_IDX = OFF                                                                                                                                             
+-------------------------------------------------------------------------------+  
```

bevanwsjones commented:
I take it you still need to update headers with the compile definitions. But lgtm

greole commented:
> I take it you still need to update headers with the compile definitions. But lgtm

Headers should be OK. I just renamed the cmake option ie 
```
if(NeoN_DEFINE_US_IDX)
  target_compile_definitions(NeoN_public_api INTERFACE NeoN_US_IDX=1)
endif()
```
It was broken earlier but i fixed it in https://github.com/exasim-project/NeoN/pull/312/commits/62babafc46aa98d3b40b95a281c5597258f250ca

HendriceH reviewed (approved):
LGTM!
